spade-core/src/main/scala/com/salesforce/mce/spade/orchard/OrchardClient.scala

```diff
@@ -22,15 +22,36 @@ class OrchardClient(setting: OrchardClient.Setting) {
 
   import setting._
 
-  def create(spadePipeline: SpadePipeline): Either[ErrorResponse, OrchardClientForPipeline] =
-    HttpRequest
-      .post[String, WorkflowRequest](
-        host.newBuilder().addPathSegment("v1").addPathSegment("workflow").build(),
-        WorkflowRequest(spadePipeline.name, spadePipeline.workflow)
-      )
-      .map(new OrchardClientForPipeline(setting, _))
+  def create(spadeWorkflowGroup: SpadeWorkflowGroup) = {
```

Comment:
with the new change, this create function can be rewritten like this

```scala
spadeWorkflowGroup.workflows.toSeq
  .map(
    HttpRequest
      .post[String, WorkflowRequest](
        host.newBuilder().addPathSegment("v1").addPathSegment("workflow").build(),
        WorkflowRequest(spadeWorkflowGroup.nameForKey(wfKey), wf)
      )
      .map { workflowId =>
        // let's not print in client
        println(s"created workflow: $workflowId")
        new OrchardClientForPipeline(setting, workflowId)
      }
  )
  .foldLeft(Right(Seq.empty)) {
    case (Right(xs), Right(x)) =>
      Right(xs :+ x)
    case (Left((err, xs)), Right(x)) =>
      Left(err, (xs :+ x))
    case (Right(xs), Left(e)) =>
      Left((e, xs))
    case (Left((err, xs)), Left(e)) =>
      Left((err, xs))
  }
```

Reply:
hmm my current way fails early at first error, without sending all requests. `.map` this way will send all requests, right?